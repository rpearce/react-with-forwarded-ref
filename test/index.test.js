/* eslint react/prop-types: 0 */

import React, { createRef } from 'react'
import { renderIntoDocument } from 'react-dom/test-utils'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import withForwardedRef from '../source'

describe('withForwardedRef', () => {
  it('renders', () => {
    const Comp = ({ children, className, forwardedRef }) => (
      <div className={className} ref={forwardedRef}>
        {children}
      </div>
    )
    const WrappedComp = withForwardedRef(Comp)
    const ref = createRef()
    const wrapper = shallow(
      <WrappedComp className="foo" ref={ref}>
        Testing 123
      </WrappedComp>
    )
    const tree = toJson(wrapper)
    expect(tree).toMatchSnapshot()
  })

  it('renders, custom displayName', () => {
    const Comp = ({ children, className, forwardedRef }) => (
      <div className={className} ref={forwardedRef}>
        {children}
      </div>
    )
    Comp.displayName = 'FooBar'
    const WrappedComp = withForwardedRef(Comp)
    const ref = createRef()
    const wrapper = shallow(
      <WrappedComp className="foo" ref={ref}>
        Testing 123
      </WrappedComp>
    )
    const tree = toJson(wrapper)
    expect(tree).toMatchSnapshot()
  })

  it('ref points to HTMLDivElement', () => {
    const Comp = ({ children, className, forwardedRef }) => (
      <div className={className} ref={forwardedRef}>
        {children}
      </div>
    )
    const WrappedComp = withForwardedRef(Comp)
    const ref = createRef()
    renderIntoDocument(
      <WrappedComp className="foo" ref={ref}>
        Testing 123
      </WrappedComp>
    )
    expect(ref.current).toBeInstanceOf(HTMLDivElement)
  })
})
