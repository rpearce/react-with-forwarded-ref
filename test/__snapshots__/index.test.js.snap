// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withForwardedRef renders 1`] = `
<Comp
  className="foo"
  forwardedRef={
    Object {
      "current": null,
    }
  }
>
  Testing 123
</Comp>
`;

exports[`withForwardedRef renders, custom displayName 1`] = `
<FooBar
  className="foo"
  forwardedRef={
    Object {
      "current": null,
    }
  }
>
  Testing 123
</FooBar>
`;
